src/python/ensembl/genes/tracking/bioproject_tracking.py

```diff
@@ -0,0 +1,250 @@
+        url = f"https://api.ncbi.nlm.nih.gov/datasets/v2alpha/taxonomy/taxon/{taxon_id}/dataset_report"
+        
+        try:
+            response = requests.get(url)
+            response.raise_for_status()  # Raises an HTTPError if the response status code is 4XX or 5XX
+            data = response.json()
+            taxonomies = data.get('reports', [])
+            for taxonomy in taxonomies:
+                rank_name = taxonomy.get('taxonomy', {}).get('classification', {}).get(rank, {}).get('name')
+                taxonomy_info = {rank : rank_name}
+                live_annotations[accession].update(taxonomy_info)
+                
+        except requests.HTTPError as http_err:
+            print(f"HTTP error occurred: {http_err}")
+        except Exception as err:
+            print(f"An error occurred: {err}")
+           
+    return(live_annotations)
+        
+def write_report(live_annotations, report_file, rank):
+    with open(Path(report_file), "w", encoding="utf8") as report_file:
+        for accession in live_annotations:
+            try:
+                report_file.write(accession + '\t' + live_annotations[accession]["dbname"] + '\t' + live_annotations[accession][rank] + '\n')
+            except TypeError:
+                report_file.write(accession + '\t' + live_annotations[accession]["dbname"] + '\t unknown \n')
+                print(live_annotations[accession])
+def main():
+    """
+    Main function to handle command-line arguments and output the result.
+    """
+    parser = argparse.ArgumentParser(description='Fetch assembly accessions from NCBI BioProject and report the number of corresponding annotations in rapid.ensembl.org.')
+    parser.add_argument('bioproject_id', type=str, help='NCBI BioProject ID')
+    parser.add_argument('--haploid', action='store_true', help='Fetch only haploid assemblies')
+    parser.add_argument('--report_file', type=str, help='Where to write report to', default='./report_file.csv')
+    parser.add_argument('--rank', type=str, help='Taxonomic rank to classify', default='order')
+    
+    args = parser.parse_args()
+
+    bioproject_accessions_taxon = get_assembly_accessions(args.bioproject_id, args.haploid)
+    live_annotations = get_ensembl_live(bioproject_accessions_taxon)
+    live_annotations_classified = get_taxonomy_info(live_annotations, bioproject_accessions_taxon, args.rank)
+    unique_taxon_ids = {details['taxon_id'] for details in live_annotations.values()}
+    print(f"Found {len(bioproject_accessions_taxon)} assemblies under BioProject ID {args.bioproject_id}")
+    print(f"Found {len(live_annotations)} annotations in rapid.ensembl.org for {len(unique_taxon_ids)} unique species")
+
+    rank_values = [details[args.rank] for details in live_annotations.values()]
+    rank_counts = Counter(rank_values)
+    print("\nBreakdown:")
+    print(rank_counts)
+    write_report(live_annotations, args.report_file, args.rank)
```

Comment:
will this report be uploaded in the new registry too?

Reply:
No definitely not. This should just be used for reporting, i.e. I ask what is the status of the CBP project, run script and pass summary to stakeholders. I think there will be aspects of what I'm doing here that will be useful for the registry and maybe elsewhere, so we might need to think about how re-use parts of the code, but the script itself has a defined usage, and that is for reporting.